doc/doxygen/headers/laoperators.h

```diff
@@ -96,6 +96,79 @@
  * InverseMatrixRichardson, SchurMatrix, ShiftedMatrix,
  * ShiftedMatrixGeneralized, TransposeMatrix
  *
+ *
+ * <h3>Packaged Operation</h3>
+ *
+ * An  application of a LinearOperator object to a vector via
+ * <code>operator*</code> yields a PackagedOperation object that stores
+ * this computation.
+ *
+ * The PackagedOperation class allows lazy evaluation of expressions
+ * involving vectors and linear operators. This is done by storing the
+ * computational expression and only performing the computation when either
+ * the object is implicitly converted to a vector object, or
+ * <code>apply</code> (or <code>apply_add</code>) is invoked by hand. This
+ * avoids unnecessary temporary storage of intermediate results.
+ *
+ * As an example consider the addition of multiple vectors:
+ * @code
+ *   dealii::Vector<double> a, b, c, d;
+ *   // ..
+ *   dealii::Vector<double> result = a + b - c + d;
+ * @endcode
+ * Converting the PackagedOperation <code>a + b - c + d</code> to a vector
+ * results in code equivalent to the following code
+ * @code
+ *   dealii::Vector<double> a, b, c, d;
+ *   // ..
+ *   dealii::Vector<double> result = a;
+ *   result += b;
+ *   result -= c;
+ *   result += d;
+ * @endcode
+ * that avoids any intermediate storage. As a second example (involving a
+ * LinearOperator object) consider the computation of a residual $b-Ax$:
+ *
+ * @code
+ *   dealii::SparseMatrix<double> A;
+ *   dealii::Vector<double> b, x;
+ *   // ..
+ *   const auto op_a = linear_operator(A);
+ *
+ *   dealii::Vector<double> residual =  b - op_a * x;
```

Comment:
I think this may not be entirely clear by itself as an example. Can you say something to the following effect:
```
Here, the expression <code>b-op_a * x</code> results in an object of type PackagedOperation that stores the <i>sequence of operations</i> that should be performed using the two vectors and the linear operation, but it does not yet actually perform these operations. However, when you assign the PackagedOperation object to the <code>residual</code> vector, the operations are actually executed (whatever the content may be of the vectors <code>b,x</code> and of the matrix <code>op_a</code> at the time of execution, as opposed to creation of the PackagedOperation object).
```